typescript/api/services/I18nEntriesService.ts

```diff
@@ -0,0 +1,453 @@
+      if (!bundle) throw new Error('Bundle not found');
+
+      const { branding, locale, namespace, id: bundleId } = bundle;
+      const data = bundle.data || {};
+      
+      // Load centralized metadata
+      const centralizedMeta = await this.loadCentralizedMeta();
+      
+      // Extract optional metadata map at root level: { [keyPath]: { category?, description? } }
+      // File-specific _meta overrides centralized meta
+      const fileMeta: Record<string, { category?: string; description?: string }> = (data && typeof data._meta === 'object') ? data._meta : {};
+      
+      // Merge centralized meta with file-specific meta (file-specific takes precedence)
+      const meta = { ...centralizedMeta, ...fileMeta };
+
+      // Flatten the data then strip any _meta entries
+      const flatAll = this.flatten(data || {});
+      const flat: Record<string, any> = {};
+      Object.keys(flatAll).forEach(k => {
+        if (k === '_meta' || k.startsWith('_meta.')) return; // skip meta keys
+        flat[k] = flatAll[k];
+      });
+      const keys = Object.keys(flat);
+
+      // Ensure we have a BrandingModel for downstream calls
+      const brandingModel: BrandingModel = (typeof branding === 'string')
+        ? (BrandingService.getBrand(branding) || ({ id: branding } as BrandingModel))
+        : (branding as BrandingModel);
+
+      // Track existing keys to detect removals
+      const existingEntries = await I18nTranslation.find({ branding: brandingModel.id || brandingModel, locale, namespace });
+      const existingKeysSet = new Set(existingEntries.map(e => e.key));
+
+      for (const key of keys) {
+        const val = flat[key];
+        existingKeysSet.delete(key); // still present
+        const existing = await this.getEntry(brandingModel, locale, namespace, key);
+        if (existing && !overwrite) continue;
+  await this.setEntry(brandingModel, locale, namespace, key, val, { bundleId, category: meta?.[key]?.category, description: meta?.[key]?.description, noReload: true });
+      }
+
+      // Any keys left in existingKeysSet are no longer present in the bundle -> remove to avoid desync
+  for (const obsoleteKey of existingKeysSet) {
+        try {
+  await this.deleteEntry(brandingModel, locale, namespace, String(obsoleteKey));
+        } catch (e) {
+          this.logger.warn('Failed to prune obsolete key', obsoleteKey, (e as Error)?.message || e);
+        }
+      }
+  // Bulk operation complete; reload once
+  try { (global as any).TranslationService?.reloadResources?.(brandingModel.id || brandingModel); } catch (_e) { /* ignore */ }
+    }
+
+
+    /**
+     * Load centralized metadata from language-defaults/meta.json
+     */
+    public async loadCentralizedMeta(): Promise<Record<string, any>> {
+      try {
+        const fs = await import('node:fs');
+        const path = await import('node:path');
+        
+        const metaPath = path.join(sails.config.appPath, 'language-defaults', 'meta.json');
```

Comment:
What's the reasoning for loading a file instead of using a sails config file?

Reply:
Added a comment to the function with more context. There's also a Readme in the language-defaults directory that explains more about its usage from a configuration perspective